### PR TITLE
Preparations for 3.16.1

### PR DESCRIPTION
**Pull request by MrAnno**
The NEWS file **does not contain** the following PRs, please review them as well:

 * Always add test datafiles to the dist tarball (#2101)
 * Loggen packaging (#2095)
 * Transport factory make type public (#2091)
 * Multitransport contains (#2087)
 * Fix test_rcptid (#2083)
 * multitransport: keep  I/O condition synced (#2082)
 * cfg: free PersistState after deallocating all LogPipe objects (#2081)
 * Use multitransport when proto needs it (#2079)
 * transport: extend TransportFactoryRegistry with remove() method (#2077)
 * Fix loggen plugin handling (#2074)
 * debuggercli: remove getframe() hacks from completerlang (#2068)
 * afsocket: add default_inet_port to protocol plugins (#2064)
 * transport-factory-registry: overwriting of a factory instance is not … (#2061)
 * multitransport: older glib support (#2060)
 * Persistent stateful client-proto support (#2059)
 * packaging/debian: Add libhook-commands.so to syslog-ng-core.install (#2041)
 * LogReader async wakeup (#1992)
 * Add missing loggen plugin files to deb package (#2024)
 * Loggen refactor fixes (#2023)
 * Fix deb package build issue caused by new loggen (#2021)
 * CMake compile issue because of PR #2003 and #2007 (#2011)
 * Ack tracker on all acked (#2104)
 * packaging/debian: remove libsyslog-ng-test.a from syslog-ng-dev package (#2115)
 * logreader: add log_reader_close_proto() (#2075)

### CODE REVIEW

File: lib/versioning.h [hunk 1]
```diff
@@ -127,11 +128,12 @@
 #define VERSION_VALUE_3_13 0x030d
 #define VERSION_VALUE_3_14 0x030e
 #define VERSION_VALUE_3_15 0x030f
+#define VERSION_VALUE_3_16 0x0310
```
Comment (MrAnno) [inline lib/versioning.h]: Is this correct?

File: NEWS.md [hunk 2]
```diff
@@ -1,68 +1,65 @@
...
- * Miscellaneous build-system related fixes and improvements (both autotools and
-   CMake).
- * Update `lib/json-c` to `json-c-0.13-20171207`. (#1900)
+ * Do not ship rabbitmq-c with syslog-ng (remove as submodule) (#2052)
+ * Add debug message to program source/destination about successful start (#2046)
+ * cfg-parser: Report memory exhaustion errors during config parsing (#2033)
+ * LogPipe: Improved debug logs (#2032)
+ * Dbld coverage (#2031)
+ * LogTransportMock enhancement (#2017)
+ * ivykis: update to 0.42.3 (#2012)
```
Comment (MrAnno) [inline NEWS.md]: move to "Notes to the developers"

File: NEWS.md [hunk 3]
```diff
@@ -1,68 +1,65 @@
...
+ * CMake modernization pt. 1 (#2007)
```
Comment (MrAnno) [inline NEWS.md]: move to "Notes to the developers" (?)

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/versioning.h (hunk 1 above)
Comment (MrAnno) [reply lib/versioning.h]: It is correct (see `lib/tests/test_lexer.c`).

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/versioning.h (hunk 1 above)
Comment (lbudai) [reply lib/versioning.h]: yes, it is correct :) (I had a patch that made it possible to use `0x0316`... but it was rejected)

Comment (MrAnno) [reply lib/versioning.h]: I've updated an assert in test_lexer, just to make sure. :smile:

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: packaging/rhel/syslog-ng.spec [hunk 4]
```diff
@@ -1,8 +1,8 @@
 %global ivykis_ver 0.36.1
 
 Name: syslog-ng
-Version: 3.15.1
-Release: 2%{?dist}
+Version: 3.16.1
+Release: 1%{?dist}
```
Comment (gaborznagy) [inline packaging/rhel/syslog-ng.spec]: @czanik: Can you review this change, please?
As we see we need to modify the `Release` part of the spec file always: `Release` is a package revision number, therefore we have to reset this number on every release.
See [rpm doc](http://ftp.rpm.org/max-rpm/s1-rpm-build-creating-spec-file.html).

Comment (czanik) [reply packaging/rhel/syslog-ng.spec]: LGTM

I changed the spec file in a way, that you only need to update the version number here, but not in the other parts of the file.

Comment (gaborznagy) [reply packaging/rhel/syslog-ng.spec]: Thanks!
I remember the previous change to this file, when the exact version number in filenames changed to a glob pattern.
https://github.com/balabit/syslog-ng/blob/b5ec5ffd4d204038319bd8521e6f4a7441fade18/packaging/rhel/syslog-ng.spec#L333-L337

**Comment by MrAnno** [comment]
@gaborznagy Thanks for the patch!

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
@lbudai Please have a look.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**